### PR TITLE
feat: free from extends BaseEntity

### PR DESCRIPTION
**Pull request by jiho-kr**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [ ] Feature
-   [x] Code style update (formatting, local variables)
-   [x] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Documentation content changes
-   [ ] Other... Please describe:

## What is the current behavior?

Issue Number: [376](https://github.com/woowabros/nestjs-library-crud/issues/376)

## What is the new behavior?

Entity does not have to `extend BaseEntity` to be used.

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: spec/genernal-entity/general.spec.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import request from 'supertest';
+import { Entity, Repository, Column, DeleteDateColumn, PrimaryGeneratedColumn, CreateDateColumn, UpdateDateColumn } from 'typeorm';
+
+import { Crud } from '../../src/lib/crud.decorator';
+import { CrudService } from '../../src/lib/crud.service';
+import { CrudController, GROUP } from '../../src/lib/interface';
+import { TestHelper } from '../test.helper';
+
+@Entity('general')
+class GeneralEntity {
```
Comment (Pigrabbit) [inline spec/genernal-entity/general.spec.ts:18]: It seems like `GeneralEntity` is duplicated in here and `spec/genernal-entity/general.entity.ts`.

File: src/lib/interface/entity.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { DeepPartial, ObjectLiteral } from 'typeorm';
+
+export type EntityType = DeepPartial<ObjectLiteral>;
```
Comment (Pigrabbit) [inline src/lib/interface/entity.ts:3]: How about just using `ObjectLiteral`?

```ts
export type EntityType = ObjectLiteral;
```

File: spec/genernal-entity/general.entity.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { ApiHideProperty, ApiProperty } from '@nestjs/swagger';
```
Comment (Pigrabbit) [inline spec/genernal-entity/general.entity.ts:1]: There is typo in directory name.
- `spec/genernal-entity/general.entity.ts` -> `spec/general-entity/general.entity.ts`

File: src/lib/interface/entity.ts (hunk 2 above)
Comment (jiho-kr) [reply src/lib/interface/entity.ts:3]: Okay, it can do without DeepPartial.

File: spec/genernal-entity/general.entity.ts (hunk 3 above)
Comment (jiho-kr) [reply spec/genernal-entity/general.entity.ts]: thanks

**Review by Pigrabbit — APPROVED** [review]

**Pull request merged**